unlock_indicator.c

```diff
@@ -667,269 +708,233 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
         }
     }
 
     // initialize positioning vars
-    double ix, iy;
-    double x, y;
-    double screen_x, screen_y;
-    double w, h;
-    double tx = 0;
-    double ty = 0;
-    double dx = 0;
-    double dy = 0;
-
-    double radius = BUTTON_RADIUS;
-    double bar_offset = 0;
-
-    int te_x_err;
-    int te_y_err;
-    // variable mapping for evaluating the clock position expression
-    te_variable vars[] = {
-        {"w", &w}, {"h", &h},
-        {"x", &screen_x}, {"y", &screen_y},
-        {"ix", &ix}, {"iy", &iy},
-        {"tx", &tx}, {"ty", &ty},
-        {"dx", &dx}, {"dy", &dy},
-        {"r", &radius}
-    };
-#define NUM_VARS 11
-
-    te_expr *te_ind_x_expr = te_compile(ind_x_expr, vars, NUM_VARS, &te_x_err);
-    te_expr *te_ind_y_expr = te_compile(ind_y_expr, vars, NUM_VARS, &te_y_err);
-    te_expr *te_time_x_expr = te_compile(time_x_expr, vars, NUM_VARS, &te_x_err);
-    te_expr *te_time_y_expr = te_compile(time_y_expr, vars, NUM_VARS, &te_y_err);
-    te_expr *te_date_x_expr = te_compile(date_x_expr, vars, NUM_VARS, &te_x_err);
-    te_expr *te_date_y_expr = te_compile(date_y_expr, vars, NUM_VARS, &te_y_err);
-    te_expr *te_layout_x_expr = te_compile(layout_x_expr, vars, NUM_VARS, &te_x_err);
-    te_expr *te_layout_y_expr = te_compile(layout_y_expr, vars, NUM_VARS, &te_y_err);
-    te_expr *te_status_x_expr = te_compile(status_x_expr, vars, NUM_VARS, &te_x_err);
-    te_expr *te_status_y_expr = te_compile(status_y_expr, vars, NUM_VARS, &te_y_err);
-    te_expr *te_modif_x_expr = te_compile(modif_x_expr, vars, NUM_VARS, &te_x_err);
-    te_expr *te_modif_y_expr = te_compile(modif_y_expr, vars, NUM_VARS, &te_y_err);
-    te_expr *te_bar_expr = te_compile(bar_expr, vars, NUM_VARS, &te_x_err);
-
-    double time_x = 0, time_y = 0,
-           date_x = 0, date_y = 0,
-           mod_x = 0, mod_y = 0,
-           layout_x = 0, layout_y = 0,
-           status_x = 0, status_y = 0;
-
-    if (screen_number < 0 || !(screen_number < xr_screens)) {
-        screen_number = 0;
-    }
+    double screen_x = 0, screen_y = 0,
+           width = 0, height = 0;
 
-    if (xr_screens > 0 && !bar_enabled) {
-        w = xr_resolutions[screen_number].width;
-        h = xr_resolutions[screen_number].height;
-        screen_x = xr_resolutions[screen_number].x;
-        screen_y = xr_resolutions[screen_number].y;
-        if (te_ind_x_expr && te_ind_y_expr) {
-            ix = 0;
-            iy = 0;
-            ix = te_eval(te_ind_x_expr);
-            iy = te_eval(te_ind_y_expr);
-        }
-        else {
-            ix = xr_resolutions[screen_number].x + (xr_resolutions[screen_number].width / 2);
-            iy = xr_resolutions[screen_number].y + (xr_resolutions[screen_number].height / 2);
+    double radius = (circle_radius + ring_width);
+    int button_diameter_physical = ceil(scaling_factor * BUTTON_DIAMETER);
+    DEBUG("scaling_factor is %.f, physical diameter is %d px\n",
```

Comment:
To figure out that weird issue with my scaling factor not being 1, I had to change this to `%f` instead of `%.f` - honestly, probably a good change to include.